src/main/java/com/google/sps/data/Chat.java

```diff
@@ -6,90 +6,133 @@
 import com.google.cloud.datastore.FullEntity;
 import com.google.cloud.datastore.Query;
 import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
 import com.google.cloud.datastore.StructuredQuery.OrderBy;
 import com.google.cloud.datastore.Key;
 import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.IncompleteKey;
+import com.google.cloud.datastore.Value;
+import com.google.cloud.datastore.ListValue;
+import com.google.cloud.datastore.StringValue;
 import com.google.gson.Gson;
 import com.google.sps.data.Chat;
 
 import java.io.IOException;
 import java.util.*;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import org.jsoup.Jsoup;
-import org.jsoup.safety.Whitelist;
 import java.lang.String;
+import java.time.LocalDateTime;
+import java.util.stream.Collectors;
 
 /** An item on a todo list. */
 public final class Chat {
 
-  private long id;
-  private String participants;//A comma separated string of participants
-  private List<String> messages;
-
-  //Constructor 1
-  Chat(long id, String participants, List<String> messages) {
-    this.id = id;
-    this.participants = participants;
-    this.messages = messages;
-  }
-
-  // //Constructor 2
-  Chat(Entity entity) {
-    this.id = (long) entity.getKey().getId();
-    this.participants = entity.getString("participants");
-    // this.messages = entity.getProperty("messages");
-  }
-
-  public static Chat getChat(Datastore datastore, String chatid) {
-    Key key = datastore.newKeyFactory().setKind("Chat").newKey(chatid);
-
-    try {
-      Entity entity = datastore.get(key);
-      Chat c = new Chat(entity);
-      return c;
-    } catch (Exception e) {
-      return null;
+    private long id;
+    private List<String> participants;
+    private List<String> messages;
+
+    // Constructor 1
+    public Chat(long id, List<String> participants, List<String> messages) {
+        this.id = id;
+        this.participants = participants;
+        this.messages = messages;
+    }
+
+    // //Constructor 2
+    public Chat(Entity entity) {
+        this.id = (long) entity.getKey().getId();
+        this.participants = entity.getList("participants").stream().map(val -> ((StringValue) val).get())
+                .collect(Collectors.toList());
+        this.messages = entity.getList("messages").stream().map(val -> ((StringValue) val).get())
+                .collect(Collectors.toList());
+    }
+
+    public static Chat newChat(Datastore datastore, String participantId1, String participantId2) {
+        Chat chat = Chat.chatExists(datastore, participantId1, participantId2);
```

Comment:
I suggest breaking these up so the code in the servlet reads a little better:
```
if (two participants):
  look up chat
  if (not present):
    create chat
  return chat
```

Reply:
Alternatively you could rename this function `getOrCreateChat` so it's better understood it'll create a chat if not found.